Java bindings generation fail

Hi @sappo 

On latest versions of Czmq & Zyre, I fail to generate Java bindings. When I run the `build.sh` script from `czmq-jni/android`, it fail on the `initCMake` task.

```
> Task :czmq-jni:initCMake FAILED
-- Checking for module 'libzmq'
--   No package 'libzmq' found
-- Could NOT find LIBZMQ (missing: LIBZMQ_LIBRARIES LIBZMQ_INCLUDE_DIRS) 
CMake Error at CMakeLists.txt:28 (message):
  libzmq not found.
```

Did I miss something ?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,11 @@ enable_testing()
 set(SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
 # Select flags
-SET(CMAKE_C_FLAGS_RELEASE "-O3")
+if(MSVC)
+    SET(CMAKE_C_FLAGS_RELEASE "/O2")
+else()
+    SET(CMAKE_C_FLAGS_RELEASE "-O3")
+endif()
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/builds/cmake/Modules)
 
@@ -84,15 +88,17 @@ file(WRITE "${PROJECT_BINARY_DIR}/platform.h.in" "
 
 configure_file("${PROJECT_BINARY_DIR}/platform.h.in" "${PROJECT_BINARY_DIR}/platform.h")
 
-#The MSVC C compiler is too out of date,
-#so the sources have to be compiled as c++
-if (MSVC)
-    enable_language(CXX)
-    file(GLOB sources "${SOURCE_DIR}/src/*.c")
-    set_source_files_properties(
-        ${sources}
-        PROPERTIES LANGUAGE CXX
-    )
+if (WIN32)
+    #The MSVC C compiler is too out of date,
+    #so the sources have to be compiled as c++
+    if (MSVC AND NOT (CMAKE_CXX_COMPILER_ID MATCHES "Clang"))
+        enable_language(CXX)
+        file(GLOB sources "${SOURCE_DIR}/src/*.c")
+        set_source_files_properties(
+            ${sources}
+            PROPERTIES LANGUAGE CXX
+        )
+    endif()
     set(MORE_LIBRARIES ws2_32 Rpcrt4 Iphlpapi)
 endif()
 

--- a/bindings/jni/README.md
+++ b/bindings/jni/README.md
@@ -6,12 +6,14 @@ JNI Binding for zyre
 
 ## Building the JNI Layer for Linux and OSX
 
-Ensure you have gradle and cmake installed, then run:
+Before you start make sure zyre is built and installed on your system.
+
+Next, ensure you have gradle and cmake installed, then run:
 
     gradle build jar
     gradle test
 
-If you don't like to install gradle beforehand just use the gradle wrapper.
+If you don't like to install gradle beforehand simply use the gradle wrapper.
 
     ./gradlew build jar
     ./gradlew test
@@ -27,6 +29,8 @@ If libraries of dependent projects are not installed in any of the default locat
     ./gradlew build jar -PbuildPrefix=/tmp/jni_build
 
 ## Building the JNI Layer for Android
+
+Before you start make sure that you've built the JNI Layer for your current OS.
 
 Please read the prerequisites section of the [README](../../builds/android/README.md) in the android build directory.
 

--- a/bindings/python_cffi/zyre_cffi/cdefs.py
+++ b/bindings/python_cffi/zyre_cffi/cdefs.py
@@ -4058,6 +4058,27 @@ void
 const char *
     zsys_ipv6_mcast_address (void);
 
+// Set IPv4 multicast address to use for sending zbeacon messages. By default
+// IPv4 multicast is NOT used. If the environment variable
+// ZSYS_IPV4_MCAST_ADDRESS is set, use that as the default IPv4 multicast
+// address. Calling this function or setting ZSYS_IPV4_MCAST_ADDRESS
+// will enable IPv4 zbeacon messages.
+void
+    zsys_set_ipv4_mcast_address (const char *value);
+
+// Return IPv4 multicast address to use for sending zbeacon, or NULL if none was
+// set.
+const char *
+    zsys_ipv4_mcast_address (void);
+
+// Set multicast TTL default is 1
+void
+    zsys_set_mcast_ttl (byte value);
+
+// Get multicast TTL
+byte
+    zsys_mcast_ttl (void);
+
 // Configure the automatic use of pre-allocated FDs when creating new sockets.
 // If 0 (default), nothing will happen. Else, when a new socket is bound, the
 // system API will be used to check if an existing pre-allocated FD with a

